src/ac_copilot_trainer/modules/setup_library.lua

```
@@ -0,0 +1,289 @@
+--- the existing schema doesn't store the setup *name* as a top-level
+--- field, we conservatively walk all files for the current car/track
+--- and surface the fastest lap with a matching setup.snapshot.path.
+---
+--- Performance: O(N) over the journal, but typical sessions have <500
+--- laps and the screen calls this per-row only when refreshing the list,
+--- not per-frame. CSP exposes `io.scanDir` + `io.open`, both blocking.
+---@param setupName string  basename without `.ini`
+---@return integer|nil  fastest lap_ms, or nil
+function M.bestForSetup(setupName)
+  if type(setupName) ~= "string" or setupName == "" then return nil end
+  if not (io and type(io.scanDir) == "function") then return nil end
```

Luacheck (1.2.0)

[warning] accessing undefined field 'scanDir' of global 'io'

(W143)